.gitignore

```diff
@@ -0,0 +1,7 @@
+CMakeCache.txt
+CMakeFiles/
+Makefile
+cmake_install.cmake
+examples/CMakeFiles/
+examples/Makefile
+examples/cmake_install.cmake
```

Comment:
I don't even know how to commit without a `.gitignore` as there are many untracked CMake generated files.

Also it's better to further add more entries into `.ignore` just like [the standard template that Github provides for node.js projects](https://github.com/github/gitignore/blob/master/Node.gitignore).